### PR TITLE
gui cont tabs: small clean-ups and improvements

### PR DESCRIPTION
**Pull request by pieleric**

### CODE REVIEW

File: src/odemis/gui/cont/tabs.py [hunk 1]
```diff
@@ -4352,8 +4353,14 @@ def _create_needed_tabs(self, tab_defs, main_frame, main_data):
...
-            else:
-                tab_def["button"].Hide()  # Hide the tab button
+                assert tab.button not in buttons
+                buttons.add(tab.button)
+
+        # Hides the buttons which are not used
+        for tab_def in tab_defs:
+            b = tab_def["button"]
+            if b not in buttons:
+                b.Hide()
```
Comment (tepals) [inline src/odemis/gui/cont/tabs.py]: Why did you move this outside the other loop?

File: src/odemis/gui/cont/tabs.py [hunk 2]
```diff
@@ -1716,6 +1716,16 @@ def __init__(self, name, button, panel, main_frame, main_data):
         self._settings_controller.setter_temporalspec_bck_file = self.set_temporalspec_background
         self._settings_controller.setter_spec_file = self.set_spec_comp
 
+        if main_data.role is None:
+            # HACK: Move the logo to inside the FileInfo bar, as the tab buttons
+            # are hidden when only one tab (ie, no backend).
```
Comment (tepals) [inline src/odemis/gui/cont/tabs.py]: when only one tab is selected or when there is only one tab?

File: src/odemis/gui/cont/tabs.py [hunk 3]
```diff
@@ -1716,6 +1716,16 @@ def __init__(self, name, button, panel, main_frame, main_data):
...
+            capbar = panel.fp_fileinfo._caption_bar
+            logo = main_frame.logo
+            assert main_frame.pnl_tabbuttons.GetSizer().Detach(logo)
+            logo.Reparent(capbar)
+            # Move to the right of the bar (before the icon)
+            logo.SetPosition((capbar.Size[0] - logo.Size[0] - 20 - 16 - 6, 8))
```
Comment (tepals) [inline src/odemis/gui/cont/tabs.py]: You could add a small comment where the  - 20 - 16 - 6 numbers are coming from

File: src/odemis/gui/cont/tabs.py (hunk 2 above)
Comment (pieleric) [reply src/odemis/gui/cont/tabs.py]: Only one tab shown. -> Fixed comment.

File: src/odemis/gui/cont/tabs.py (hunk 1 above)
Comment (pieleric) [reply src/odemis/gui/cont/tabs.py]: Because now buttons can be shared (see commit message). So we cannot immediately hide a button if the tab is not shown. We have to check that no tab uses this button.

File: src/odemis/gui/cont/tabs.py (hunk 3 above)
Comment (pieleric) [reply src/odemis/gui/cont/tabs.py]: Indeed, that looks a little too magical... -> added extra comment.
